lib/src/rules/non_constant_identifier_names.dart

```diff
@@ -43,11 +43,14 @@ class NonConstantIdentifierNames extends LintRule implements NodeLintRule {
   void registerNodeProcessors(
       NodeLintRegistry registry, LinterContext context) {
     var visitor = _Visitor(this);
+    registry.addCatchClause(this, visitor);
     registry.addConstructorDeclaration(this, visitor);
+    registry.addForEachPartsWithDeclaration(this, visitor);
     registry.addFormalParameterList(this, visitor);
     registry.addFunctionDeclaration(this, visitor);
     registry.addMethodDeclaration(this, visitor);
     registry.addVariableDeclaration(this, visitor);
```

Comment:
I think this will flag non-local variables (such as top-level variables and fields), but it's hard to imagine that nobody has ever run into this bug before.

On the other hand, I think you need to visit `ForPartsWithDeclarations`. `visitVariableDeclaration` is probably what's causing that test to pass.

Reply:
> I think this will flag non-local variables (such as top-level variables and fields)

Right.  And we *do* want to flag those.  IOW: feature not  bug?

Reply:
Oh, right. I'd gotten it in my head that this lint was for local variables. But it's only this PR that's about local variables.

Reply:
Thanks for the extra look!